### PR TITLE
Use `spring.jpa.properties.hibernate.ejb.naming_strategy` instead of …

### PR DESCRIPTION
**Pull request by izeye**
…default if it exists.

### CODE REVIEW

**Comment by philwebb** [comment]
Can you provide some more information here. What's issue is this pull request trying to address? Why inline the `getHibernateNamingStrategy` method?

**Comment by izeye** [comment]
@philwebb I think `spring.jpa.properties.hibernate.ejb.naming_strategy` should be used instead of default value when a user specifies it. I added a test which shows my intention. Am I missing something?

**Comment by philwebb** [comment]
It's better for us if you can add some comments to the pull request/issue, it's sometimes hard to follow the reasoning if we need to look at the code.

I think I get it. If the user specifies an `ejb.naming_strategy` directly, we should back off?

**Comment by philwebb** [comment]
Possibly related to #3149

**Comment by izeye** [comment]
@philwebb Yes, I meant "back off". Sorry that I didn't give enough information. I'll try to be better next time. Thanks for the feedback :+1:

**Comment by snicoll** [comment]
I don't think it's related to #3149 FWIW.

**Comment by philwebb** [comment]
It appears that the existing behavior is intentional; changing it break `HibernateJpaAutoConfigurationTests.testCustomNamingStrategyViaJpaProperties`.

**Comment by izeye** [comment]
@philwebb Oh, sorry. I read the comment in the test and I see it's intentional now.

I looked into the code and thought it's a potential bug because it doesn't back off.

Sorry again for a false alarm.
